docs/signing-spec.md

```diff
@@ -19,6 +19,52 @@ For example, by signing a parcel's hash, we can in effect certify that _an entit
 
 Asymmetric cryptography is nice for this because a signer can distribute a public key and a signature, and that is enough information for someone to verify that the signature was generated by the signer's private key.
 
+## Signing and Roles
+
+Signers of an invoice MUST have a role. The following roles are defined by this specification.
+Agents SHOULD initiate a warning and MAY initiate a failure if another role is encountered.
+
+- `creator`: Signer asserts that they are the ones who made this bindle
+- `proxy`: Signer asserts that they are a consumer or waypoint for this bindle, and have no reason to distrust
+- `host`: Signer asserts that it has served as a Bindle host for this content
+
+The `creator` signature MUST be on any invoice that is compliant with this system.
+There MAY be multiple creators on an invoice.
+A creator MUST NOT sign the same invoice with the same key multiple times.
+A Bindle server MUST reject an invoice that is not signed by a creator.
+A Bindle server MUST reject an invoice that is signed by a creator's key for which the server does not have an entry in its keyring.
+A client SHOULD reject an invoice signed by a creator whose public key is not in the client's keyring
+
+Bindle servers MUST add a `host` entry to a bindle upon receiving the bindle from a client.
+A bindle server MUST NOT sign the same invoice more than once with the same key.
+Clients MAY reject invoices that are signed by unknown hosts, but they also MAY allow them.
+
+Proxy signatures are designed for provenance more than security. When a non-creator pushes
+a bindle to a Bindle server, the non-creator SHOULD sign the invoice with a proxy signature.
+
+Examples of proxy signers:
+- CI systems
+- Users who transfer bindles from one Bindle server to another
+- Import/export tools
+
+## Signing and Timestamps
+
+The date and time of a signature can be an important piece of information for auditing.
+
+Both invoice and parcel signatures require an `at` field with a UNIX timestamp expressed as an unsigned 64-bit integer with no leading zeros.
+
+> Time stamps MUST be greater than 0. It is nonsensical to have a signature older than the the specification. For the sake of consistency, though, a 0 value is allowed in the formatting. However, implementations MAY reject any entries whose timestamps predate Sept. 1, 2020.
+
+Since signature blocks are not primarily designed for human consumption, an implementation should prioritize simplicity and consistency over readability.
+The UNIX timestamp offers this:
+
+- It is well defined
+- It is adopted broadly
+- It is an unambiguous format that can easily be represented, serialized, and parsed
+- It is accurate enough for the problem at hand
+
+Longer textual formats are subject to parsing and serialization ambiguities, as well in quirks of implementation across languages.
+
 ## Signing Parcels
```

Comment:
So just for my understanding, if someone wants to sign parcels, they'll need to do it when they create the invoice, correct? The label is only present within the invoice and not separate as presented below. 

Nit: I'd recommend showing the label inside of the context of an invoice, since labels are no longer standalone

Reply:
I am going to have to rewrite the parcels signing stuff. RIght now, it does not make sense because it is based on the older labeled model. And signing it might not be necessary now if we don't have external labels. I'll get this in a follow-up when I can think through it as its own case.